x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/CheckTargetShardsCountStepTests.java

```diff
@@ -88,4 +118,47 @@ public void testStepIncompleteIfTargetShardsCountNotValid() {
             )
         );
     }
+
+    public void testStepWithInvalidAction() {
+        String indexName = randomAlphaOfLength(10);
+        String policyName = "test-ilm-policy";
+        CheckTargetShardsCountStep checkTargetShardsCountStep = new CheckTargetShardsCountStep(randomStepKey(), randomStepKey());
+
+        IndexMetadata indexMetadata = IndexMetadata.builder(indexName)
+            .settings(settings(Version.CURRENT).put(LifecycleSettings.LIFECYCLE_NAME, policyName))
+            .numberOfShards(10)
+            .numberOfReplicas(randomIntBetween(0, 5))
+            .build();
+
+        Phase phase = new Phase(
+            checkTargetShardsCountStep.getKey().getPhase(),
+            TimeValue.ZERO,
+            singletonMap(checkTargetShardsCountStep.getKey().getAction(), new ReadOnlyAction())
+        );
+        LifecyclePolicy lifecyclePolicy = new LifecyclePolicy(policyName, singletonMap(phase.getName(), phase));
```

Comment:
NIT: why not use `Map.of(` throughout (here and in the other usage of the singleton map below) like you also used below to keep things consistent? :)

Reply:
I will, sloppy copy paste ;)

Reply:
Fixed